plugin-gradle/src/main/java/com/diffplug/gradle/spotless/FormatExtension.java

```diff
@@ -443,8 +388,8 @@ protected void setupTask(SpotlessTask task) {
 		task.setExceptionPolicy(exceptionPolicy);
 		task.setTarget(target);
 		task.setSteps(steps);
-		if (licenseHeaderExtension != null) {
-			licenseHeaderExtension.setupTask(task);
+		if (licenseHeaderConfig != null) {
+			licenseHeaderConfig.setupTask(this, task);
 		}
 		task.setLineEndingsPolicy(getLineEndings().createPolicy(getProject().getProjectDir(), () -> task.target));
```

Comment:
Order matters for FormatterStep.  e.g. if I have a license with trailing whitespace, I'll get a different result if I trim whitespace then apply the license,  vs apply the license then trim whitespace.  Currently, the order that you declare steps is the order that they all happen (which is the rationale behind the weird "replaceStep()" in the examples I linked.  This approach allows you to not have a licenseHeaderConfig object.

After fixing this,  just squash the commits and update the CHANGES.md and we should be good to go.

Reply:
Ok, I've updated the PR to include the CHANGES and fully follow the existing "config object pattern" including the replaceStep() logic.

Again thanks for the feedback and very impressive response times.